charts/node/templates/statefulset.yaml

```diff
@@ -212,6 +233,28 @@ spec:
             - mountPath: /var/run/secrets/{{ .type }}
               name: {{ .type }}
           {{- end }}
+        {{ else if .Values.node.vault.keys }}
+        - name: inject-vault-keys
+          image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
+          command: [ "/bin/sh" ]
+          args:
+            - -c
+            - |
+              {{- range $keys := .Values.node.vault.keys }}
+              {{ $.Values.node.command }} key insert --base-path /data \
+              --chain ${CHAIN} \
+              --key-type $(awk '/{{ .typeTarget }}/ {print $2}' /vault/secrets/{{ .name }}) \
+              --scheme $(awk '/{{ .schemeTarget }}/ {print $2}' /vault/secrets/{{ .name }}) \
+              --suri "$(awk '/{{ .seedTarget }}/ { $1=""; print}' /vault/secrets/{{ .name }} | cut -c 2-)" \
```

Comment:
These `awk` commands could definitely use someone with better bash-fu than myself sanity checking them.

Reply:
If possible, I would prefer to avoid complex awk queries if the secret can be properly saved to a temporary file. It would also help not having to template/parse the same secrets twice (first in the go template then with awk commands).